docs/src/modules_schema.md

```diff
@@ -37,15 +37,42 @@ list of (submodule)
 
 - [modules/commands.nix](https://github.com/numtide/devshell/tree/main/modules/commands.nix)
 
-### `commands.*.package`
+### `commands.*`
+
+A config for a command when the `commands` option is a list.
+
+**Type**:
+
+```console
+(package or string convertible to it) or (list with two elements of types: [ string (package or string convertible to it) ]) or (flatOptions)
+```
```

Comment:
For flake-parts users, this mix of types would make it hard to merge commands coming from different places. I would rather keep either the list, or the attrset of commands, but not both.

Let's keep things simple.

Reply:
I couldn't invent a new option name so I made `commands` either a list or an attrset.

I prefer attrsets with attrNames as command group names.

I think it should be the first-class feature and should be documented and type checked. 

That's why, I declined the idea to provide just a function that converts an attrset to a list.

So, I see three approaches:
1. Allow `commands` only as an `attrset`
1. Create a new option for commands as an attrset, leave commands as a `list`
  - what should be the name?
1. Allow `commands` to be either a `list` or an `attrset`, do something about merging lists and attrsets for flake-parts users.
  - I need a concrete example

Reply:
I had no problems with flake-parts. See [repro](https://github.com/deemp/gists/tree/9300b80e4b39ca8ca647d94c115a0984dcb186b8/devshellFlakeParts)